projects/ngx-monaco-tree/src/lib/monaco-tree-file/monaco-tree-file.component.ts

```diff
@@ -70,142 +155,48 @@ export class MonacoTreeFileComponent implements OnChanges {
-			}
-
-		}
-	}
+  }
 
-	toggle(shouldEmit = true) {
-		this.open = !this.open;
-    if (shouldEmit) {
-		  this.clickFile.emit(this.name)
+  toggle(shouldChangeCurrent = true) {
+    this.open.update((open) => !open);
+    if (shouldChangeCurrent) {
+      this.current.set(this.fullPath());
     }
-	}
-
-	get style() {
-		return 'margin-left: '+ 10*this.depth +'px'
-	}
-
-	get folder() {
-		return this.content !== null && this.content !== undefined
-	}
-
-  get isActive() {
-    return this.current === this.path;
   }
 
-	handleClickFile(file: string) {
-		this.clickFile.emit(this.name + '/' + file);
-	}
-
-	handleRightClickFile(event: MouseEvent) {
-		event.preventDefault()
-		const pos = getAbsolutePosition(event.target);
-		this.position = [pos.x + event.offsetX, pos.y + event.offsetY]
-	}
-
-	handleRightClick(event: ContextMenuAction) {
-		this.contextMenuClick.emit([event[0], this.name + '/' + event[1]]);
-	}
+  handleRightClickFile(event: MouseEvent) {
+    event.preventDefault()
+    const pos = getAbsolutePosition(event.target);
+    this.position.set([pos.x + event.offsetX, pos.y + event.offsetY]);
+  }
 
   collapseAll() {
     this.children().forEach((child) => child.collapseAll());
-    this.open = false;
+    this.open.set(false);
   }
 
-	@HostListener('document:contextmenu', ['$event'])
-	clickOut(event: MouseEvent) {
-		if(!this.eRef.nativeElement.contains(event.target)) {
-			this.position = [-1000, -1000];
-		}
-	}
-
-  get colorStyle() {
-    switch(this.color) {
-      case 'red':
-        return '#c74e39'
-      case 'yellow':
-        return '#e2c08d'
-      case 'green':
-        return '#81b88b'
-      case 'gray':
-        return '#8c8c8c'
-      default:
-        if(this.color?.startsWith("#")) return this.color;
-        else if(this.color) {
-          console.warn("Invalid color ", this.color, " please use red | yellow | green | gray or a valid hex color with #.")
-          return null;
-        } else {
-          return null;
-        }
+  @HostListener('document:contextmenu', ['$event'])
+  clickOut(event: MouseEvent) {
+    if (!this.eRef.nativeElement.contains(event.target)) {
+      this.position.set([-1000, -1000]);
     }
   }
 
   drop($event: CdkDragDrop<any>) {
     const file = $event.item.data;
-    //Find the container where the file is dropped (thank copilot)
+    // Find the container where the file is dropped (thank copilot)
     const containers = document.querySelectorAll('.monaco-tree-file-container');
-    let targetContainer: Element|null = null;
-    for (const container of Array.from(containers)) {
-      const boundingRect = container.getBoundingClientRect();
-      if ($event.dropPoint.x >= boundingRect.left && $event.dropPoint.x <= boundingRect.right &&
-        $event.dropPoint.y >= boundingRect.top && $event.dropPoint.y <= boundingRect.bottom) {
-        targetContainer = container;
-      }
-    }
-
+    const targetContainer = Array.from(containers).find((container) => {
```

Comment:
not sure about the impact but it seems the previous implementation was using the latest match whereas the new implementation is returning the first match